test/type.coffee

```diff
@@ -8,19 +8,33 @@ Amen.describe "Type functions", (context) ->
     isString, isFunction, isObject, isArray, isRegExp, isDate, isDefined,
     isGenerator, isPromise} = require "../src/type"
 
-  context.test "type"
-
-  context.test "isType"
-
-  context.test "instanceOf"
+  context.test "type", ->
+    assert type 7
+    assert type "7"
+    assert type true
+
+  context.test "isType", ->
+    assert isType(String, "7")
+    assert isType(Number, 7)
+    assert isType(Boolean, true)
+    assert ! isType(Number, false)
+
+  context.test "instanceOf", ->
+    as = -> isType()
+    bs = -> as 7
+    assert bs instanceOf as
 
   context.test "isNumber", ->
     assert isNumber 7
     assert ! isNumber "7"
     assert ! isNumber false
 
-  context.test "isNaN"
-  context.test "isFinite"
+#isNaN seems buggy
```

Comment:
`NaN` is kinda funny.  It means "Not a Number".  It's JavaScript's way of saying that it has failed in some operation that requires or returns real numbers.  So, it only crops up in limited cases.

```coffee
assert isNaN Math.sqrt "foobar"   # Can't operate on a String.
assert isNaN Math.sqrt -1            # JavaScript only works with real numbers, not imaginary 
assert isNaN 0 / 0                        # This is considered indeterminate, so this is how JavaScript says there's no good real number answer.
```